book/invalidation.md

```diff
@@ -167,6 +167,79 @@ API!
 
 [editing-hates-you]: https://lord.io/text-editing-hates-you-too/
 
+
+Why Invalidation?
+=================
+
+Fundamentally, the reason editing this page is slow in our browser is
+that this page is pretty big.[^other-reasons] After all, it's not
+handling the keypress that's slow---that just appends a character to a
+`Text` node, which takes almost no time. It's what happens afterwards
+that takes time: the browser has to rerender the whole page. Even tiny
+changes take a long time on a large page.
+
+[^other-reasons]: I'm sure there are all sorts of performance
+    improvements possible without implementing the invalidation
+    techniques from this chapter, but they are essential toward
+    achieving incrementalism, which is a kind of asymptotic guarantee
+    that simple optimizations won't achieve.
+
+If we want interactions to be fast, even on large, complex pages,
+we're going to need re-rendering the page to take time *proportional
+to the size of the change*, not proportional to the size of the page.
+This property is sometimes called *incrementality*, and it really
+constrains our browser. For example, incrementality means that during
+rendering, we can't even *traverse* the whole page, since that would
+take time proportional to the whole page, not the change being made.
+
+Now, note that when the page is first loaded, rendering is definitely
+going to take time proportional to the size of the page. So to achieve
+incrementality, we're going to need to think of the initial render and
+later re-renders differently. The initial render will create a cache
+that later renders invalidate and recompute parts of.[^big-change] If
+we're only invalidate a small part each time, this will only take time
+proportional to the changed portion of the page.
+
+[^big-change]: While initial and later renders are conceptually
+    different, they'll use the same code path. Basically, the initial
+    render will be one big change from no page to the initial page,
+    while later re-renders will incorporate smaller changes.
+
+In a real browser, every step of the rendering pipeline needs to be
+incremental, but this chapter focuses on layout. Layout is important
+and makes a good illustration of the idea. Incrementalizing layout
+means skipping the two most expensive parts of layout: building the
+layout tree and traversing it to compute layout fields. Ultimately,
+re-layouts will take under a millisecond when typing on this page.
+
+Getting there is going to be a bit of a rocky road, though. The whole
```

Comment:
Instead of stating generalities about the hard work to come, being more explicit about it?

Something like:

"We'll have to proceed with the following tasks:
* Make a detailed *dependency graph* of what depends on what, down to the level of every
value (`width`, `height`, `y`, etc) on every layout object.
* Use that graph to determine what to recompute when something changes.